dags/mlflow-multimodel-dag.py

```diff
@@ -0,0 +1,274 @@
+        """
+
+        
+        # Onehot encoding 
+        df = pd.get_dummies(df, prefix='workclass', columns=['workclass'])
+        df = pd.get_dummies(df, prefix='education', columns=['education'])
+        df = pd.get_dummies(df, prefix='occupation', columns=['occupation'])
+        df = pd.get_dummies(df, prefix='race', columns=['race'])
+        df = pd.get_dummies(df, prefix='sex', columns=['sex'])
+        df = pd.get_dummies(df, prefix='income_bracket', columns=['income_bracket'])
+        df = pd.get_dummies(df, prefix='native_country', columns=['native_country'])
+
+
+        # Bin Ages
+        df['age_bins'] = pd.cut(x=df['age'], bins=[16,29,39,49,59,100], labels=[1, 2, 3, 4, 5])
+
+
+        # Dependent Variable
+        df['never_married'] = df['marital_status'].apply(lambda x: 1 if x == 'Never-married' else 0) 
+
+
+        # Drop redundant colulmn
+        df.drop(columns=['income_bracket_<=50K', 'marital_status', 'age'], inplace=True)
+
+        return df
+
+
+    @task_group(group_id='grid_search_cv')
+    def grid_search_cv(features):
+        """Train and validate model
+        
+        Returns accuracy score via XCom to GCS bucket.
+
+        Keyword arguments:
+        df -- data from previous step pulled from BigQuery to be processed. 
+        """
+
+        model_grid_params = { 
+            'lgbm':{
+                'learning_rate': [0.01, .05, .1], 
+                'n_estimators': [50, 100, 150],
+                'num_leaves': [31, 40, 80],
+                'max_depth': [16, 24, 31, 40],
+                'boosting_type': ['gbdt'], 
+                'objective': ['binary'],
+                'seed': [55]
+                },
+            'log_reg':{
+                'penalty': ['l1','l2','elasticnet'],
+                'C': [0.001, 0.01, 0.1, 1, 10, 100],
+                'solver': ['newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga']
+                }
+            }
+
+
+        models = [
+            ['lgbm', lgb.LGBMClassifier(objective='binary', metric=['auc', 'binary_logloss'])],
+            ['log_reg', LogisticRegression(max_iter=500)]
+        ]
+
+        tasks = []
+
+        for model in models:
+            @task(task_id=model[0])
+            def train(df: pd.DataFrame, model_type=model[0],model=model[1], grid_params=model_grid_params[model[0]], **kwargs):
+
+                y = df['never_married']
+                X = df.drop(columns=['never_married'])
+
+
+                X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=55, stratify=y)
+                
+                mlflow.set_tracking_uri('http://host.docker.internal:5000')
+                try:
+                    # Creating an experiment 
+                    mlflow.create_experiment('census_prediction')
+                except:
+                    pass
+                # Setting the environment with the created experiment
+                mlflow.set_experiment('census_prediction')
```

Comment:
Why doesn't this overwrite the previous model's experiment when the for loop index increases?